packages/cloudflare-pages/src/cloudflare-pages.ts

```diff
@@ -25,14 +32,38 @@ export const defaultOptions: Required<CloudflarePagesOptions> = {
   external: [],
   minify: true,
   emptyOutDir: false,
+  routesJson: true,
 }
 
+type StaticRoutes = { version: number; include: string[]; exclude: string[] }
+
 export const cloudflarePagesPlugin = (options?: CloudflarePagesOptions): Plugin => {
   const virtualEntryId = 'virtual:cloudflare-pages-entry-module'
   const resolvedVirtualEntryId = '\0' + virtualEntryId
+  let config: ResolvedConfig
+  let staticRoutes: StaticRoutes
+  const staticPaths: string[] = []
 
   return {
     name: '@hono/vite-cloudflare-pages',
+    configResolved: async (resolvedConfig) => {
+      config = resolvedConfig
+      const paths = await readdir(config.publicDir, {
+        withFileTypes: true,
+      })
+      paths.forEach((p) => {
+        if (p.isDirectory()) {
+          staticPaths.push(`/${p.name}/*`)
+        } else {
+          staticPaths.push(`/${p.name}`)
+        }
+      })
+      staticRoutes = {
+        version: 1,
+        include: ['/*'],
+        exclude: staticPaths,
+      }
```

Comment:
> This approach is fine for minimal projects, but we can't use no more than 100 include/exclude rules combined.

You are right. One of the best practices is to create a directory like `/static` on the top level because it will be written as `/static/*`. We can write about it on README.